src/sst/core/config.cc

```diff
@@ -41,163 +43,826 @@ using namespace std;
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+        struct tm res = {}; /* This warns on GCC 4.8 due to a bug in GCC */
+#pragma GCC diagnostic pop
+        char* p;
+
+        for ( size_t i = 0; i < n_templ; i++ ) {
+            memset(&res, '\0', sizeof(res));
+            p = strptime(arg.c_str(), templates[i], &res);
+            fprintf(
+                stderr, "**** [%s]  p = %p ; *p = '%c', %u:%u:%u\n", templates[i], p, (p) ? *p : '\0', res.tm_hour,
+                res.tm_min, res.tm_sec);
+            if ( p != nullptr && *p == '\0' ) {
+                cfg.stopAfterSec_ = res.tm_sec;
+                cfg.stopAfterSec_ += res.tm_min * 60;
+                cfg.stopAfterSec_ += res.tm_hour * 60 * 60;
+                return true;
+            }
+        }
+
+        fprintf(
+            stderr,
+            "Failed to parse stop time [%s]\n"
+            "Valid formats are:\n",
+            arg.c_str());
+        for ( size_t i = 0; i < n_templ; i++ ) {
+            fprintf(stderr, "\t%s\n", templates[i]);
+        }
+
+        return false;
+    }
+
+    // timebase
+    bool setTimebase(const std::string& arg)
+    {
+        /* TODO: Error checking */
+        cfg.timeBase_ = arg;
+        return true;
+    }
+
+    // partitioner
+    bool setPartitioner(const std::string& arg)
+    {
+        cfg.partitioner_ = arg;
+        if ( cfg.partitioner_.find('.') == cfg.partitioner_.npos ) { cfg.partitioner_ = "sst." + cfg.partitioner_; }
+        return true;
+    }
+
+    // heart beat
+    bool setHeartbeat(const std::string& arg)
+    {
+        /* TODO: Error checking */
+        cfg.heartbeatPeriod_ = arg;
+        return true;
+    }
+
+    // output directory
+    bool setOutputDir(const std::string& arg)
+    {
+        cfg.output_directory_ = arg;
+        return true;
+    }
+
+    // core output prefix
+    bool setOutputPrefix(const std::string& arg)
+    {
+        cfg.output_core_prefix_ = arg;
+        return true;
+    }
+
+
+    // Configuration output
+
+    // output config python
+    bool setWriteConfig(const std::string& arg)
+    {
+        cfg.output_config_graph_ = arg;
+        return true;
+    }
+
+    // output config json
+    bool setWriteJSON(const std::string& arg)
+    {
+        cfg.output_json_ = arg;
+        return true;
+    }
+
+    // parallel output
+#ifdef SST_CONFIG_HAVE_MPI
+    bool enableParallelOutput()
+    {
+        // If this is only a one rank job, then we can ignore
+        if ( cfg.world_size_.rank == 1 ) return true;
+
+        cfg.parallel_output_  = true;
+        // For parallel output, we always need to output the partition
+        // info as well
+        cfg.output_partition_ = true;
+        return true;
+    }
+
+    bool enableParallelOutputArg(const std::string& arg)
+    {
+        // If this is only a one rank job, then we can ignore, except
+        // that we will turn on output_partition
+        if ( cfg.world_size_.rank == 1 ) return true;
```

Comment:
Do you need to turn on output_partition before returning?

Reply:
No, the help message says that the option is ignored for single rank jobs.